WalletWasabi.Fluent/ViewModels/Login/LoginViewModel.cs

```diff
@@ -67,7 +69,26 @@ private async Task OnNextAsync(ClosedWalletViewModel closedWalletViewModel, Wall
 
 		if (legalResult)
 		{
-			LoginWallet(closedWalletViewModel);
+			var km = closedWalletViewModel.Wallet.KeyManager;
+
+			// This should be impossible, this is just a sanity check
+			if (!km.IsCoinjoinProfileSelected && km.AutoCoinJoin)
+			{
+				DialogResult<bool> isProfileSelected = await NavigateDialogAsync(new CoinJoinProfilesViewModel(closedWalletViewModel.Wallet.KeyManager, false), NavigationTarget.DialogScreen);
+
+				if (isProfileSelected.Result)
+				{
+					LoginWallet(closedWalletViewModel);
+				}
+				else
+				{
+					wallet.Logout();
+				}
+			}
+			else
+			{
+				LoginWallet(closedWalletViewModel);
+			}
```

Comment:
It would be worth extracting this somehow to a method since there are so many if-else here. Ignore it if you don't agree.

Reply:
I agree, I just couldn't find a clean way to do it.